### PR TITLE
feat(account-lib): avalanche C implement transactionBuilder, transferBuilder, tests

### PR DESCRIPTION
**Pull request by shiksb**
TICKET: STLX-5079

### CODE REVIEW

**Review by CapnMigraine — COMMENTED** [review]
Please rebase to fix the merge conflicts

**Review by smaria95 — CHANGES_REQUESTED** [review]

File: modules/account-lib/src/coin/avaxc/walletUtil.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// TODO: add method ids
+export const sendMultisigMethodId = '';
+export const sendMultisigTokenMethodId = '';
+export const createForwarderMethodId = '';
+export const walletInitializationFirstBytes = '';
+export const flushForwarderTokensMethodId = '';
+export const flushCoinsMethodId = '';
```
[the remark this answers is not in the document]
Comment (karankurbur) [reply modules/account-lib/src/coin/avaxc/walletUtil.ts:7]: Account lib is only used for ETH-like coins and not for ETH. ETH-like only support v2 so the method ids should match the v2 interface

[the remark this answers is not in the document]
Comment (smaria95) [reply modules/account-lib/src/coin/avaxc/walletUtil.ts:7]: @karankurbur, @CapnMigraine 
I did some extra research on this and I found that `uint` is actually an alias for `uint256`, for that reason the current `methodId` is working for the v2 contracts. 

https://docs.soliditylang.org/en/v0.5.3/types.html#integers


Also, I tested this PR changes with a script on sandbox. 
I managed to initialize a wallet and also send a transfer using accountLib to create the transaction and sending it with RPC Method `eth_sendRawTransaction` (which is the one used by indexer to broadcast eth-like transactions).

**Review by smaria95 — DISMISSED** [review]
LGTM. 
Please remember to rebase and squash your branch.
Once you do that, you can ask for my re-review and I'll approve it again

File: modules/account-lib/src/coin/avaxc/resources.ts [hunk 2]
```diff
@@ -1,13 +1,30 @@
+import EthereumCommon from '@ethereumjs/common';
+import { coins, EthereumNetwork } from '@bitgo/statics';
+
 /**
  * A Common object defining the chain
+ * TODO BG-35011: create custom Common object since Avalanche doesn't have hardforks
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/avaxc/resources.ts:6]: Moved to the STLX project (https://bitgoinc.atlassian.net/browse/STLX-5737)

File: modules/account-lib/src/coin/avaxc/walletUtil.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// TODO BG-35012: determine if and how to use methodIds
```
Comment (CapnMigraine) [inline modules/account-lib/src/coin/avaxc/walletUtil.ts:1]: Moved to the STLX project (https://bitgoinc.atlassian.net/browse/STLX-5738)

**Review by CapnMigraine — APPROVED** [review]

**Review by smaria95 — APPROVED** [review]
LGTM

**Review by CapnMigraine — APPROVED** [review]

**Pull request merged**